text/0012-CLI-commands.md

```diff
@@ -0,0 +1,1361 @@
+  -p, --no-publish        do not publish this dataset to the registry
+      --secrets strings   transform secrets as comma separated key,value,key,value,... sequence
+  -t, --title string      title of commit message for save
+```
+
+
+
+<a id='qri_search'></a>
+## qri search
+
+Search qri nodes for datasets.
+
+### Synopsis
+
+Search datasets & peers that match your query. Search pings the qri registry. Any dataset 
+that has been published to the registry is available for search.
+
+Search will match datasets based on the meta title, keyword, and theme.
+
+```
+qri search [flags] <search term>
+```
+
+### Examples
+
+```
+  # search 
+  $ qri search "annual population"
+```
+
+### Options
+
+```
+  -f, --format string   set output format [json]
+  -h, --help            help for search
+```
+
+
+
+<a id='qri_setup'></a>
+## qri setup
+
+initialize qri and IPFS repositories, provision a new qri ID
+
+### Synopsis
+
+
+Setup is the first command you run to get a fresh install of Qri. If you’ve 
+never run qri before, you’ll need to run setup before you can do anything. 
+
+Setup does a few things:
+- create a qri repository to keep all of your data
+- provisions a new qri ID
+- create an IPFS repository if one doesn’t exist
+
+This command is automatically run if you invoke any Qri command without first 
+running setup. If setup has already been run, by default Qri won’t let you 
+overwrite this info.
+
+Use the `--remove` to remove your Qri repo. This deletes your entire repo, 
+including all your datasets, and de-registers your peername from the registry.
+
+```
+qri setup [flags]
+```
+
+### Examples
+
+```
+  run setup with a peername of your choosing:
+  $ qri setup --peername=your_great_peername
+```
+
+### Options
+
+```
+  -a, --anonymous            use an auto-generated peername
+      --config-data string   json-encoded configuration data, specify a filepath with '@' prefix
+  -h, --help                 help for setup
+      --init-ipfs            initialize an IPFS repo if one isn't present (default true)
+      --ipfs-config string   json-encoded configuration data, specify a filepath with '@' prefix
+      --overwrite            overwrite repo if one exists
+      --peername string      choose your desired peername
+      --registry string      override default registry URL
+      --remove               permanently remove qri, overrides all setup options
+```
+
+
+
+<a id='qri_use'></a>
+## qri use
+
+Select datasets for use with other commands
+
+### Synopsis
+
+Select datasets for use with other commands. 
+
+Run the `use` command to 'save' a reference to a specific dataset. This dataset
+will be referenced for future commands, if no dataset reference is explicitly
+given for those commands.
+
+We created this command to ease the typing/copy and pasting burden while using
+Qri to explore a dataset.
+
+Dataset references saved using the `use` command will work with:
+
+body
+diff
+export
+get
+info
+log
+render
+validate
+
```

Comment:
I am asserting this to be true. I believe it was the original goal. Currently, `use` only works correctly with get.

Reply:
❤️ . I think we should leave it here for now, and in the future consider another RFC that adds the dataset editing commands to `use`, but requires only one dataset be selected:
- save
- remove
- rename

Reply:
I actually didn't realize that it was possible to `use` multiple datasets at a time. Right now the text is written in the singular case everywhere: "remember a reference to a specific dataset. This dataset will..." as opposed to "remember references to a specific group of datasets. These datasets will..."

Definitely like `--add` to use an additional dataset. In addition, we could change `qri use <ref>` to `qri use <refs>` to set multiple refs at once. Still `rename` should not recognize `use`, since it's the only command where ordering has an important semantic meaning.

Reply:
Yes, totally my fault for the confusion.

Currently, `qri use` only works for `qri get`. If you look at the `qri use` documentation as it currently exists in the app, it speaks in the plural. You can `get` info on multiple datasets at once.

I was always a little suspect of having multiple references, and my bias came out when writing this, hahah. 

I assume that remembering multiple references is what we want, as that is how it is written currently, and also how `qri get` functions.

Reply:
yep, big fan of both multiple refs and `--add`. I vote for ratifying all this in the current RFC, and ensuring the docs refer to multiple `<refs>`.

THE RFC PROCESS IS PROVIDING CLARITY.
(plz read that in [this guy's voice](https://youtu.be/8UFIrA7hvio?t=21))